EKF/SensorRangeFinder.cpp

```diff
@@ -32,94 +32,86 @@
+void SensorRangeFinder::runChecks(const uint64_t time_delayed_us, const Dcmf &R_to_earth)
+{
+	// calculate 2,2 element of rotation matrix from sensor frame to earth frame
+	// this is required for use of range finder and flow data
+	_R_rng_to_earth_2_2 = R_to_earth(2, 0) * _sin_tilt_rng + R_to_earth(2, 2) * _cos_tilt_rng;
 
-	_dt_last_range_update_filt_us = fminf(_dt_last_range_update_filt_us, 4e6f);
+	updateRangeDataValidity(time_delayed_us);
 }
 
-void Ekf::updateRangeDataValidity()
+void SensorRangeFinder::updateRangeDataValidity(uint64_t time_delayed_us)
 {
-	updateRangeDataContinuity();
+	updateRangeDataContinuity(time_delayed_us);
 
 	// check if out of date
-	if ((_imu_sample_delayed.time_us - _range_sample_delayed.time_us) > 2 * RNG_MAX_INTERVAL) {
+	if (((time_delayed_us - _range_sample_delayed.time_us) > 2 * RNG_MAX_INTERVAL)
+	    || !isRangeDataContinuous()) {
 		_rng_hgt_valid = false;
 		return;
 	}
 
-	// Don't allow faulty flag to clear unless range data is continuous
-	if (!_rng_hgt_valid && !isRangeDataContinuous()) {
-		return;
-	}
+	// Don't run the checks unless we have retrieved new data from the buffer
+	if (_range_data_ready) {
+		_rng_hgt_valid = false;
 
-	// Don't run the checks after this unless we have retrieved new data from the buffer
-	if (!_range_data_ready) {
-		return;
-	}
+		if (_range_sample_delayed.quality == 0) {
+			_time_bad_rng_signal_quality = time_delayed_us;
 
-	if (_range_sample_delayed.quality == 0) {
-		_time_bad_rng_signal_quality = _imu_sample_delayed.time_us;
-		_rng_hgt_valid = false;
-	} else if (_imu_sample_delayed.time_us - _time_bad_rng_signal_quality > (unsigned)_params.range_signal_hysteresis_ms) {
-		_rng_hgt_valid = true;
-	}
+		} else if (time_delayed_us - _time_bad_rng_signal_quality > _range_signal_hysteresis_ms) {
+			const bool is_in_range = ((_range_sample_delayed.rng > _rng_valid_min_val)
+						  && (_range_sample_delayed.rng < _rng_valid_max_val));
 
-	// Check if excessively tilted
-	if (_R_rng_to_earth_2_2 < _params.range_cos_max_tilt) {
-		_rng_hgt_valid = false;
-		return;
-	}
+			if (isTiltOk() || is_in_range) {
+				updateRangeDataStuck();
 
-	// Check if out of range
-	if ((_range_sample_delayed.rng > _rng_valid_max_val)
-	|| (_range_sample_delayed.rng < _rng_valid_min_val)) {
-		if (_control_status.flags.in_air) {
-			_rng_hgt_valid = false;
-			return;
-		} else {
-			// Range finders can fail to provide valid readings when resting on the ground
-			// or being handled by the user, which prevents use of as a primary height sensor.
-			// To work around this issue, we replace out of range data with the expected on ground value.
-			_range_sample_delayed.rng = _params.rng_gnd_clearance;
-			return;
+				if (!_is_stuck) {
+					_rng_hgt_valid = true;
+				}
+			}
 		}
 	}
+}
 
-	updateRangeDataStuck();
+void SensorRangeFinder::updateRangeDataContinuity(uint64_t time_delayed_us)
+{
+	// update range data continuous flag (1Hz ie 2000 ms)
```

Comment:
These comments are out of date. I would replace them with something like:

// Calculate a first order IIR low-pass filtered time of arrival between samples using a 2 second time constant.
// Apply spike protection to the filter state.
// Timing is in micro seconds.